### PR TITLE
Clone only the relevant commit w/ depth=1 on install

### PR DESCRIPTION
**Pull request by charlieparkes**

### CODE REVIEW

File: bin/install.sh [hunk 1]
```diff
@@ -4,31 +4,21 @@ set -e
...
-  (
-    cd $BUILD_HARNESS_PROJECT && \
-    git fetch origin --tags --force && \
-    git checkout $BUILD_HARNESS_BRANCH >/dev/null && \
-    git pull \
-  )
-else
-  echo "Cloning from ${GITHUB_REPO}#${BUILD_HARNESS_BRANCH}"
-  git clone -b $BUILD_HARNESS_BRANCH $GITHUB_REPO >/dev/null 2>&1
+if [ -z "$BUILD_HARNESS_BRANCH" ] || [ "$BUILD_HARNESS_BRANCH" == "master" ] ; then
```
Comment (jtdoepke) [inline bin/install.sh:9]: How would we clone the latest from master when developing the build-harness then?

Comment (charlieparkes) [reply bin/install.sh:9]: I was thinking we could just always test from a branch. Goals:

1. User can pin to a release/tag
2. User can pin to a tag older than the currently checked out version without running `clean`
2. User can pin to a branch (other than master)

I think this covers all the necessary use cases?

Comment (charlieparkes) [reply bin/install.sh:9]: (You can also still test master HEAD locally by just checking out master into your current working path.)

Comment (jtdoepke) [reply bin/install.sh:9]: Good point.

**Review by jtdoepke — APPROVED** [review]

**Pull request merged**